src/components/floating-menu/floating-menu.js

```diff
@@ -169,6 +183,7 @@ class FloatingMenu extends mixin(createComponent, eventedShowHideState) {
       }
       this._getContainer().appendChild(this.element);
       this._place();
+      (this.element.querySelector(this.options.selectorPrimaryFocus) || this.element).focus();
```

Comment:
> My concern is that if a dev is dynamically loading or changing the menu items and doesn't properly add selectorPrimaryFocus then the end user's experience will be extremely broken

I can see the concern. There is a logic to alleviate that - I have made a change to the sample HTML to demonstrate how it works: https://github.com/carbon-design-system/carbon-components/pull/219/commits/230d50009016a7eae75542fc5e06a539c33003a4

You can remove `data-floating-menu-primary-focus` attributes from the HTML to see that.